docs/user/architecture.rst

```diff
@@ -0,0 +1,83 @@
+
+.. image:: _static/img/sampleQueryArch.png
+   :scale: 75%
+   :align: center
+
+Key-value stores and Z-curves
+-----------------------------
+
+The data stores that GeoMesa uses for long-term storage are key-value databases, a type of NoSQL database in which every record is stored and retrieved using a unique identifier for that record known as a key. Accumulo, HBase, and Google Cloud Bigtable sort these keys and can store them across any number of nodes (servers). 
+
+When using a key-value database, good design of the keys themselves can lead to more efficient applications. Unlike relational databases, where the keys are frequently sequential integers, key value stores usually use the key to represent a feature by which the data are frequently queried. For example, imagine a database of customer orders being indexed by the order number. Then, when a client queries by order number, the database goes directly to that key and returns the value—that is, the record.
+
+This is a simplification of how Accumulo, HBase, and Cloud Bigtable key structures actually work, but the foundational principle of GeoMesa can be explained in terms of keys and values. To store spatio-temporal data, we need to create a key that represents the time/space location of the record. GeoMesa uses this system to store locations as points along a special line that visits all the sectors of a map, like the red line shown here: 
+
+.. image:: _static/img/Zcurve-LoRes.png
+   :scale: 50%
+   :align: center
+
+This red line is known as a space-filling curve, or to be more specific, a Z-Curve. This line visits each cell exactly once, establishing a unique ordering of cells.
+
+Space-filling curves can also work with higher-resolution maps, like the one shown here:
+
+.. following 53% instead of 50 because the image was a little smaller than the one above
+.. image:: _static/img/Zcurve-HiRes.png
+   :scale: 53%
+   :align: center
+
+Each point in this curve can be assigned a sequential value, letting GeoMesa represent what might have been a latitude-longitude pair as a single integer. This is great for representing two-dimensional data in a one-dimensional key, as is the case with a key-value datastore. Even more significantly, these space-filling curves can be adapted for n dimensions, letting GeoMesa linearize  n dimensions of data in a single dimension. 
+
+GeoMesa’s Index
+---------------
+
+The basic principle of GeoMesa’s index is to represent the three dimensions of longitude, latitude, and time with a three-dimensional space filling curve, using the values of the points along this curve as the key. This lets it store a record in a key-value store with a key that represents the three data dimensions that we most often use for queries. 
+
+The actual key structure is more complex than a simple key-value pair. Below is a more detailed representation of GeoMesa's index in Accumulo: 
+
+.. image:: _static/img/GM_Index.png
```

Comment:
this image isn't accurate anymore... the key is:
row - 2 byte epoch week, 8 bytes z3, feature ID (doesn't have to be a UUID... maybe say unique ID)
column family - a constant ('F')
column qualifier - empty
value - byte encode simple feature (i would remove the word lazy)
timestamp and viz are as indicated